prov/gni/src/gnix_msg.c

```diff
@@ -1044,6 +1050,8 @@ static int __smsg_rndzv_fin(void *data, void *msg)
 				"GNI_SmsgRelease returned %s\n",
 				gni_err_str[status]);
 		ret = gnixu_to_fi_errno(status);
+	} else {
+		ret = gnixu_to_fi_errno(hdr->status);
```

Comment:
The error status should determine what kind of event is sent.  We need some logic like this:

```
if (hdr->status not success)
 	__gnix_msg_send_completion(ep, req);
else 
        __gnix_msg_send_err(ep, req)
```

__gnix_msg_send_err() should update the correct counter and generate the error event.